### PR TITLE
Google Map

### PR DESCRIPTION
**Pull request by justasm**
- Add Google Maps
- Drop Mapbox
- Fix crash on orientation change
- Make active story radius configurable via remote config `story_radius_m`
- Design tweaks

screenshot_20161119-180528

### CODE REVIEW

File: app/src/main/java/lt/gyvosistorijos/MainController.kt [hunk 1]
```diff
@@ -1,44 +1,49 @@
...
-    internal lateinit var locationServices: LocationServices
+    internal lateinit var map: GoogleMap
+    internal lateinit var locationService: LocationService
 
     internal var zoomedIn: Boolean = false
     internal var activeStory: Story? = null
 
-    internal val storyMarkers: MutableList<MarkerView> = ArrayList()
-    internal val markerIdToStory = LongSparseArray<Story>()
+    internal val storyMarkers: MutableList<Marker> = ArrayList()
```
Comment (vycius) [inline app/src/main/java/lt/gyvosistorijos/MainController.kt:45]: In Kotlin can be simplified to `mutableListOf<Marker>()`

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
-        mapbox:center_latitude="54.6872"
-        mapbox:center_longitude="25.2797"
-        mapbox:logo_enabled="false"
-        mapbox:style_url="@string/MAPBOX_MAP_STYLE_URL"
-        mapbox:zoom="11"
+        map:cameraTargetLat="54.6872"
+        map:cameraTargetLng="25.2797"
+        map:cameraZoom="11"
+        map:uiTiltGestures="false"
+        map:uiRotateGestures="false"
```
Comment (vycius) [inline app/src/main/res/layout/activity_main.xml:18]: I see no reason to disable rotate and tilt gestures

File: app/src/main/java/lt/gyvosistorijos/MainActivity.kt [hunk 3]
```diff
@@ -1,77 +1,112 @@
...
-            map = mapboxMap
+            map.uiSettings.isMyLocationButtonEnabled = false
 
-            map.uiSettings.isTiltGesturesEnabled = false
-            map.uiSettings.isRotateGesturesEnabled = false
+            initMapStyle()
 
+            val router = Conductor.attachRouter(this, controller_container, savedInstanceState)
             if (!router.hasRootController()) {
                 router.setRoot(RouterTransaction.with(SyncController()))
```
Comment (vycius) [inline app/src/main/java/lt/gyvosistorijos/MainActivity.kt:50]: Not related to this PR, but why are we attaching router only on map loaded ?

**Review by vycius — APPROVED** [review]
Fantastic. Merging this now and releasing. I will take a walk with this in the city today.

File: app/src/main/java/lt/gyvosistorijos/MainActivity.kt (hunk 3 above)
Comment (justasm) [reply app/src/main/java/lt/gyvosistorijos/MainActivity.kt:50]: `MainController` currently assumes there's a map to use. So it's either this or propagate the map loaded event down via some mechanism.

**Pull request merged**